setup.py

```diff
@@ -0,0 +1,143 @@
+"""
+setuptools-based build of ROOT.
+
+This script uses setuptools API to steer a custom CMake build of ROOT. All build
+configuration options are specified in the class responsible for building. The
+installation step produces the following output target install directory:
+
+```
+site-packages/
+-- cppyy/
+-- cppyy_backend/
+-- libcppyy.so
+-- libcppyy_backend.so
+-- ROOT/
+-- -- libROOTPythonizations.so
+```
+
+A custom extension module is injected in the setuptools setup to properly
+generate the wheel with CPython extension metadata.
+"""
+
+from setuptools import setup, find_packages, Extension
+import pathlib
+import tempfile
+from setuptools.command.build import build as _build
+from setuptools.command.install import install as _install
+
+import subprocess
+import os
+import shlex
+
+# Get the long description from the README file
+SOURCE_DIR = pathlib.Path(__file__).parent.resolve()
+LONG_DESCRIPTION = (SOURCE_DIR / "README.md").read_text(encoding="utf-8")
+
+BUILD_DIR = tempfile.mkdtemp()
+INSTALL_DIR = tempfile.mkdtemp()
+
+# Name given to an internal directory within the build directory
+# used to mimick the structure of the target installation directory
+# in the user Python environment, usually named "site-packages"
+ROOT_BUILD_INTERNAL_DIRNAME = "mock_site_packages"
+
+
+class ROOTBuild(_build):
+    def run(self):
+        _build.run(self)
+
+        # Configure ROOT build
+        configure_command = shlex.split(
+            "cmake -GNinja -Dccache=ON "
+            "-Dgminimal=ON -Dasimage=ON -Dopengl=OFF "  # Graphics
+            "-Druntime_cxxmodules=ON -Drpath=ON -Dfail-on-missing=ON -DROOT_WHEEL_BUILD=ON "  # Generic build configuration
```

Comment:
```suggestion
            "-Druntime_cxxmodules=ON -Drpath=ON -Dfail-on-missing=ON "  # Generic build configuration
```
Maybe with my suggestions, this is not needed anymore?